tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -0,0 +1,206 @@
+
+static struct bt_codec lc3_codec =
+	BT_CODEC_LC3(BT_CODEC_LC3_FREQ_ANY, BT_CODEC_LC3_DURATION_10,
+		     BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1),
+		     40u, 120u, 1u,
+		     (BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL |
+		     BT_AUDIO_CONTEXT_TYPE_MEDIA));
+
+static struct bt_pacs_cap cap_sink = {
+	.codec = &lc3_codec,
+};
+
+static struct bt_pacs_cap cap_source = {
+	.codec = &lc3_codec,
+};
+
+static int set_location(void)
+{
+	int err;
+
+	err = bt_pacs_set_location(BT_AUDIO_DIR_SINK,
+				   BT_AUDIO_LOCATION_FRONT_CENTER);
+	if (err != 0)
+		return err;
+
+	err = bt_pacs_set_location(BT_AUDIO_DIR_SOURCE,
+				   (BT_AUDIO_LOCATION_FRONT_LEFT |
+				   BT_AUDIO_LOCATION_FRONT_RIGHT));
+	if (err != 0)
+		return err;
+
+	return 0;
+}
+
+static int set_available_contexts(void)
+{
+	int err;
+
+	err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SOURCE,
+					     AVAILABLE_SOURCE_CONTEXT);
+	if (err != 0)
+		return err;
+
+	err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SINK,
+					     AVAILABLE_SINK_CONTEXT);
+	if (err != 0)
+		return err;
+
+	return 0;
+}
+
+static int set_supported_contexts(void)
+{
+	int err;
+
+	err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SOURCE,
+					     SUPPORTED_SOURCE_CONTEXT);
+	if (err != 0)
+		return err;
+
+	err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SINK,
+					     SUPPORTED_SINK_CONTEXT);
+	if (err != 0)
+		return err;
+
+	return 0;
+}
+
+static void pacs_supported_commands(uint8_t *data, uint16_t len)
+{
+	uint8_t cmds[2];
+	struct pacs_read_supported_commands_rp *rp = (void *) cmds;
+
+	(void)memset(cmds, 0, sizeof(cmds));
+
+	tester_set_bit(cmds, PACS_READ_SUPPORTED_COMMANDS);
+
+	tester_send(BTP_SERVICE_ID_PACS, PACS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, (uint8_t *) rp, sizeof(cmds));
+}
+
+static void pacs_update_characteristic(uint8_t *data, uint16_t len)
+{
+	int err = 0;
+	uint8_t status = BTP_STATUS_SUCCESS;
+	const struct pacs_update_characteristic_cmd *cmd = (void *) data;
+
+	switch (cmd->characteristic) {
+	case PACS_CHARACTERISTIC_SINK_PAC:
+		err = bt_pacs_cap_unregister(BT_AUDIO_DIR_SINK,
+					     &cap_sink);
+		break;
+	case PACS_CHARACTERISTIC_SOURCE_PAC:
+		err = bt_pacs_cap_unregister(BT_AUDIO_DIR_SOURCE,
+					     &cap_source);
+		break;
+	case PACS_CHARACTERISTIC_SINK_AUDIO_LOCATIONS:
+		err = bt_pacs_set_location(BT_AUDIO_DIR_SINK,
+				BT_AUDIO_LOCATION_FRONT_CENTER |
+				BT_AUDIO_LOCATION_BACK_CENTER);
+		break;
+	case PACS_CHARACTERISTIC_SOURCE_AUDIO_LOCATIONS:
+		err = bt_pacs_set_location(BT_AUDIO_DIR_SOURCE,
+				(BT_AUDIO_LOCATION_FRONT_LEFT |
+				BT_AUDIO_LOCATION_FRONT_RIGHT |
+				BT_AUDIO_LOCATION_FRONT_CENTER));
+		break;
+	case PACS_CHARACTERISTIC_AVAILABLE_AUDIO_CONTEXTS:
+		err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SOURCE,
+				BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+		break;
+	case PACS_CHARACTERISTIC_SUPPORTED_AUDIO_CONTEXTS:
+		err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SOURCE,
```

Comment:
Are the values to set here not from BTP?

Reply:
PAC server test cases are very simple and checks only if notification was received, not its content. @sjanc suggested me to wait with larger extension of the PACS BTP until PTS will require specific configuration to avoid multiple changes of the BTP.

Reply:
Alright,so we just want to see a different value here to trigger a notification. Thanks :)